### PR TITLE
add support for transit gateway vpc attachment

### PR DESCRIPTION
**Pull request by ivan-savciuc**

### CODE REVIEW

**Comment by rikonen** [comment]
I haven't checked the server side implementation but is there anything in the properties returned via the API that distinguishes transit gateway from regular peering connection? Was just thinking if e.g. import should check that the resource is in fact transit gateway vpc attachment.

**Comment by ivan-savciuc** [comment]
The only difference is `user_peer_network_cidrs` in comparison with regular VPC peering connection. My understanding is that Aiven is using the same API endpoint for both types but logically distinguishes between them by `user_peer_network_cidrs` if it present in the request and response we are talking about VPC attachment.

File: aiven/resource_vpc_peering_connection.go [hunk 1]
```diff
@@ -262,6 +269,12 @@ func copyVPCPeeringConnectionPropertiesFromAPIResponseToTerraform(
 		return err
 	}
 
+	if len(peeringConnection.UserPeerNetworkCIDRs) > 0 {
+		if err := d.Set("user_peer_network_cidrs", peeringConnection.UserPeerNetworkCIDRs); err != nil {
+			return err
+		}
+	}
```
Comment (rikonen) [inline aiven/resource_vpc_peering_connection.go:276]: Should this not clear old cidrs in case list is empty?

**Comment by ivan-savciuc** [comment]
And there is a separate API endpoint to manage CIDR`s after https://api.aiven.io/doc/#operation/Project.VpcPeeringConnectionUpdate after VPC creation

File: aiven/resource_transit_gateway_vpc_attachement.go [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+			}
+		}
+
+		if forDeletion {
+			deleteCIDRs = append(deleteCIDRs, old)
+		}
+	}
+
+	_, err = client.TransitGatewayVPCAttachment.Update(projectName, vpcID, aiven.TransitGatewayVPCAttachmentRequest{
+		Add:    nil,
```
Comment (rikonen) [inline aiven/resource_transit_gateway_vpc_attachement.go:131]: Should be `Add: add`?

File: aiven/resource_vpc_peering_connection.go (hunk 1 above)
Comment (ivan-savciuc) [reply aiven/resource_vpc_peering_connection.go:276]: Indeed, I was trying here to distinguish between Gateway Attachment and regular VPC by checking this parameter empty, but it introduces this problem.

**Pull request merged**